src/components/PopupWithForm.js

```diff
@@ -20,6 +20,13 @@ export class PopupWithForm extends Popup {
     return this._form;
   }
 
+  setButtonText(text) {
+    const submitButton = this._form.querySelector("[type='submit']");
+    if (submitButton) {
+      submitButton.textContent = text;
+    }
+  }
+
   setEventListeners() {
     this._form = this._popup.querySelector(".form");
 
```

Comment:
Please, delete `this._form.reset();` from line 37 in `PopupWithForm`


You need to **reset** the form only   in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.